lib/grape/middleware/base.rb

```diff
@@ -21,8 +22,38 @@ def call(env)
       def call!(env)
         @env = env
         before
+
         @app_response = @app.call(@env)
-        after || @app_response
+
+        # Merge Headers from API into @header
```

Comment:
Lets optimize this a bit. Default `@header` to `nil`, then extract this part into a method and `return unless @header`. Assign `@header` the first time inside `header` with `@header ||= {}`. 

Also lets clear the ambiguity between `header` and `headers`. I don't think they should be aliased, maybe `header` should only be a function that sets a header instead of returning `@header`?